src/main/java/com/meilisearch/sdk/json/GsonJsonHandler.java

```diff
@@ -13,37 +10,21 @@
 import com.meilisearch.sdk.model.Key;
 
 public class GsonJsonHandler implements JsonHandler {
-    private Gson gson;
 
-    public GsonJsonHandler() {
-        this.gson = new Gson();
-    }
+    private final Gson gson;
 
-    public GsonJsonHandler(Gson gson) {
-        this.gson = gson;
+    public GsonJsonHandler() {
+        GsonBuilder builder = new GsonBuilder();
+        builder.registerTypeAdapter(Key.class, new GsonKeyTypeAdapter());
```

Comment:
Can we do that "automatically" only when are about to serialize a Key?

Reply:
We could, but that would mean doing a check in the `encode(Object o)` to ensure the type of `o` is `Key`. I figured that one of the primary objectives was to remove the key specific checks and logic.

The primary purpose of a "type adapter", is that we can tell Gson, that when it comes to encoding or decoding a specific type (in this case, `Key.class` as the first argument), we can control what happens during that process. We're not really affecting how the rest of the Gson library works for any type, other than `Key`.

Reply:
Oh, I got it! That's cool :D